model/inline/src/main/scala/aqua/model/inline/TagInliner.scala

```diff
@@ -371,7 +378,16 @@ object TagInliner extends Logging {
         } yield model.fold(TagInlined.Empty())(m => TagInlined.Single(model = m))
 
       case RestrictionTag(name, typ) =>
-        pure(RestrictionModel(name, typ))
+        // restriction can be generated before inlining and not renamed,
+        // rename it after inlined children with new exports
+        State.pure(TagInlined.After(() => {
+          for {
+            exps <- Exports[S].exports
+            model = exps.get(name).collect { case VarModel(n, _, _) =>
+              TagInlined.Single[S](RestrictionModel(n, typ))
+            }
```

Comment:
I kinda believe we should reuse `ValueModel#resolveWith` here, but I am not sure